offix/src/main/java/org/aerogear/offix/interceptor/ConflictInterceptor.kt

```diff
@@ -40,6 +49,7 @@ class ConflictInterceptor(private val conflictResolutionImpl: ConfliceResolution
     inner class OffixConflictCallback(val conflictResolutionImpl: ConfliceResolutionInterface) :
         ApolloInterceptor.CallBack {
         private val TAG = javaClass.simpleName
+        private lateinit var userCallback: ApolloInterceptor.CallBack
```

Comment:
This won't work in case of onFailure, since at a given time only either onResponse or onFailure will be called.
Instead of making this a lateinit var, simply initialize it here.
```
val userCallback = queueCallback.removeFirst()
```

And then user it in both onResponse and onFailure.

Reply:
Oh yes!